src/app/teacher/abstract-tag-dialog.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Directive, OnInit } from '@angular/core';
+import {
+  FormControl,
+  Validators,
+  ValidatorFn,
+  AbstractControl,
+  ValidationErrors
+} from '@angular/forms';
+import { ProjectTagService } from '../../assets/wise5/services/projectTagService';
+import { Tag } from '../domain/tag';
+
+@Directive()
+export abstract class AbstractTagDialogComponent implements OnInit {
+  protected tagControl = new FormControl('', [
+    Validators.required,
+    this.createUniqueTagValidator()
+  ]);
+  protected tags: Tag[] = [];
+
+  constructor(protected projectTagService: ProjectTagService) {}
+
+  ngOnInit(): void {
+    this.projectTagService.retrieveUserTags().subscribe((tags: Tag[]) => {
+      this.tags = tags;
+    });
+  }
+
+  private createUniqueTagValidator(): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      return this.tags != null && this.doesTagAlreadyExist(this.tags, control.value)
```

Comment:
Remove null check and only pass in the new tagText to doesTagAlreadyExist()?
```suggestion
      return this.doesTagAlreadyExist(control.value)
```